src/efftester.ml

```diff
@@ -1551,7 +1542,14 @@ let rec tcheck env term =
          (typ, no_eff)
        with Failure msg -> failwith msg)
     | Error msg -> failwith msg)
-  | PatternMatch _ -> failwith "FIXME: tcheck: pat match not implemented"
+  | PatternMatch (typ, _matched_trm, cases, eff) ->
+    (* how to ensure that patterns can be applied to the matched term? *)
```

Comment:
One way to ensure this would be to define an `imm_type_pat` function, that works like `imm_type` but for pattern. That would require adding extra parameters in the constructors of the `pattern` type, to store those types.